sql/src/test/java/io/crate/integrationtests/JoinIntegrationTest.java

```diff
@@ -679,4 +680,45 @@ public void testGlobalAggregateMultiTableJoin() throws Exception {
         execute("select sum(t1.val), avg(t2.id), min(t3.id) from t1 inner join t2 on t1.t2 = t2.id inner join t3 on t2.t3 = t3.id");
         assertThat(TestingHelpers.printedTable(response.rows()), is("3.689999930560589| 2.0| 1\n"));
     }
+
+    @Test
+    public void testJoinWithAggregationGroupBy() throws Exception {
+        createColorsAndSizes();
+        execute("select colors.name, count(colors.name) from colors, sizes group by colors.name order by colors.name DESC");
+        assertThat(TestingHelpers.printedTable(response.rows()), is("red| 2\ngreen| 2\nblue| 2\n"));
+    }
+
+    @Test
+    public void testJoinWithGroupByLimitAndOffset() throws Exception {
+        createColorsAndSizes();
+        execute("select colors.name from colors, sizes group by colors.name order by colors.name DESC limit 1 offset 1");
+        assertThat(TestingHelpers.printedTable(response.rows()), is("green\n"));
+    }
+
+    @Test
+    public void testJoinWithGroupByAndHaving() throws Exception {
+        createColorsAndSizes();
+        execute("select colors.name from colors, sizes group by colors.name having count(colors.name) < 1;");
+        assertThat(TestingHelpers.printedTable(response.rows()), is(""));
+    }
+
+    @Test
+    public void testJoinWithGroupByAndWhere() throws Exception {
+        createColorsAndSizes();
+        execute("select colors.name from colors, sizes where colors.name='red' group by colors.name");
+        assertThat(TestingHelpers.printedTable(response.rows()), is("red\n"));
+    }
+
+    @Test(expected = SQLActionException.class)
```

Comment:
usually we prefer using a `expectedException` rule, which is used before the statement which is expected to throw the exception